lib/Data/Decimate.pm

```diff
@@ -0,0 +1,203 @@
+package Data::Decimate;
+
+use strict;
+use warnings;
+
+use 5.010;
+
+use Exporter qw/import/;
+
+our @EXPORT_OK = qw(decimate);
+
+=head1 NAME
+
+Data::Decimate - A module that allows to decimate a data feed. 
+
+=head1 VERSION
+
+Version 0.01
+
+=head1 SYNOPSIS
+
+  use Data::Decimate qw(decimate);
+
+  my @data = (
+        {epoch  => 1479203101,
+        ...},
+        {epoch  => 1479203102,
+        ...},
+        {epoch  => 1479203103,
+        ...},
+        ...
+        {epoch  => 1479203114,
+        ...},
+        {epoch  => 1479203117,
+        ...},
+        {epoch  => 1479203118,
+        ...},
+        ...
+  );
+
+  my $output = Data::Decimate::decimate(15, \@data);
+
+  #epoch=1479203114 , decimate_epoch=1479203115
+  print $output->[0]->{epoch};
+  print $output->[0]->{decimate_epoch};
+
+=head1 DESCRIPTION
+
+A module that allows you to resample a data feed
+
+=cut
+
+our $VERSION = '0.01';
+
+=head1 SUBROUTINES/METHODS
+=cut
+
+=head2 decimate
+
+Decimate a given data based on sampling frequency.
+
+=cut
+
+sub decimate {
+    my ($interval, $data) = @_;
+
+    if (not defined $interval or not defined $data or ref($data) ne "ARRAY") {
+        die "interval and data are required parameters.";
+    }
+
+    my @res;
+    my $el = $data->[0];
+    my $de = do {
+        use integer;
+        (($el->{epoch} + $interval - 1) / $interval) * $interval;
```

Comment:
@tfoertsch , In the previous code we were using modulus to calculate decimate epoch. In your suggestion , you come up with the above way. May I know what is the benefit? Is it speed ?